File: backend/app/models/driver.py
from datetime import date
from typing import Optional
from pydantic import Field
from app.models.base import CamelModel


class BookingRequest(CamelModel):
    station_id: str = Field(min_length=1)
    booking_date: date
    start_time: Optional[str] = None
    user_lat: Optional[float] = None
    user_lng: Optional[float] = None


class DriverLocation(CamelModel):
    name: str
    lat: float
    lng: float


class DriverFilterTag(CamelModel):
    id: str
    label: str


class DriverStatusOption(CamelModel):
    value: str
    label: str


class DriverLegendItem(CamelModel):
    status: str
    label: str


class DriverVehicleTypeOption(CamelModel):
    value: str
    label: str


class BookingConfig(CamelModel):
    service_fee: int = Field(ge=0)
    time_slots: list[str]
    slot_duration_minutes: int = Field(ge=15, le=360)


class DriverConfig(CamelModel):
    location: DriverLocation
    location_label: str
    search_radius_km: float = Field(gt=0)
    display_radius_km: float = Field(gt=0)
    personalized_label: str
    search_placeholder: str
    filter_tags: list[DriverFilterTag]
    status_options: list[DriverStatusOption]
    vehicle_type_options: list[DriverVehicleTypeOption]
    legend: list[DriverLegendItem]
    booking: BookingConfig

File: frontend/src/types/driver.ts
import type { Station, StationStatus } from '@/types';

export interface DriverLocation {
  name: string;
  lat: number;
  lng: number;
}

export interface DriverFilterTag {
  id: string;
  label: string;
}

export interface DriverStatusOption {
  value: string;
  label: string;
}

export interface DriverVehicleTypeOption {
  value: string;
  label: string;
}

export interface DriverLegendItem {
  status: StationStatus | string;
  label: string;
}

export interface DriverBookingConfig {
  serviceFee: number;
  timeSlots: string[];
  slotDurationMinutes: number;
}

export interface DriverConfig {
  location: DriverLocation;
  locationLabel: string;
  searchRadiusKm: number;
  displayRadiusKm: number;
  personalizedLabel: string;
  searchPlaceholder: string;
  filterTags: DriverFilterTag[];
  statusOptions: DriverStatusOption[];
  vehicleTypeOptions: DriverVehicleTypeOption[];
  legend: DriverLegendItem[];
  booking: DriverBookingConfig;
}

export interface AvailabilitySummary {
  available: number;
  busy: number;
  offline: number;
  total: number;
}

export interface LocationSearchResult {
  label: string;
  lat: number;
  lng: number;
  availability: AvailabilitySummary;
}

export interface TripStop {
  label: string;
  lat: number;
  lng: number;
  source: string;
}

export interface TripPlanRequest {
  stops: TripStop[];
  vehicleType?: string;
  corridorKm?: number;
}

export interface TripLeg {
  fromLabel: string;
  toLabel: string;
  distanceKm: number;
  durationMin: number;
}

export interface TripRoute {
  distanceKm: number;
  durationMin: number;
  polyline: string;
  bbox: number[];
  legs: TripLeg[];
}

export interface RouteStationInfo {
  station: Station;
  distanceToRouteKm: number;
  distanceFromStartKm: number;
  etaFromStartMin: number;
  estimatedChargeMin: number;
  capacityPorts: number;
  availablePorts: number;
}

export interface TripPlanResponse {
  route: TripRoute;
  stations: RouteStationInfo[];
}
